hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -186,8 +188,10 @@ define([
           _.extend(entity.health, revisionDetails.health);
           _.extend(entity.pay, revisionDetails.pay);
           _.extend(entity.pension, revisionDetails.pension);
-          _.each(entity.leave, function (leaveType, leaveTypeId) {
-            _.extend(leaveType, revisionDetails.leave ? revisionDetails.leave[leaveTypeId] : '');
+          _.each(entity.leave, function (leaveType) {
+            _.extend(leaveType, revisionDetails.leave
+              ? _.find(revisionDetails.leave, { leave_type: leaveType.leave_type })
```

Comment:
how different is this compared to 
```js
newScope.leave ? _.find(newScope.leave, { leave_type: leaveType.leave_type }) || contractRevisionIdObj : contractRevisionIdObj
```
isn't it basically the same except for the fallback value, here it's `''` and before it's `contractRevisionIdObj`

Reply:
I think it is dangerous to amend the existing code at the moment. I try to modify as less as possible now.